packages/backend/src/plugins/linguist.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TaskScheduleDefinition } from '@backstage/backend-tasks';
+import { createRouter } from '@backstage/plugin-linguist-backend';
+import { Router } from 'express';
+import type { PluginEnvironment } from '../types';
+
+export default async function createPlugin(
+  env: PluginEnvironment,
+): Promise<Router> {
+
+  const schedule: TaskScheduleDefinition = {
```

Comment:
I've been torn on this: should I create a new type/object to wrap this along with the `age` that in the `app-config.yaml` or leave it as is? Could use some advice

Reply:
Alright, would love to have a discussion here about the best way to schedule this work. You've made some really good comments that are worth working through. The comments you made are:

- Then the question is, how do we schedule the actual worker best. If it only gets the chance to inspect the work queue once per day as is the case now, it may have many thousands of things to update in one fell swoop. For that to succeed, it probably will have to be very very careful internally to throttle itself during that run, so that it doesn't exhaust the rate limiting. Could it make sense to schedule it instead, like, every minute? And then work off the top of the queue a little if there's anything to do at all at that point. Not sure.

- Could get interesting if the service is scaled up to a large number of hosts I guess. That's when a work queue that you try to atomically pick individual items off of can be beneficial; then at least you don't run the risk of multiple workers fetching the same items in parallel.

Based on these if it's scheduled to run say every 2-5 minutes what would the process look like? I was thinking something like this:

1. Get all the entities from the Catalog and add any new entities that aren't already in the Linguist database
2. Proceed with the processing the top _n_ (this would be configurable) entities based on a timestamp in the database.

For 1 do I need to worry about this taking a long time? Like if someone has a really large catalog won't I need to fetch everything? In my mind all I really need is the entity ref, I won't be storing anything else, well the linguist data later on.

For 2 how would I got about making this work where other workers could potentially grab chunks from the "queue"? Is there any example of this you could point me to?

Reply:
@freben, though I'd like to merge this in and iterate I think I need to resolve the above before we do that. Could I get some guidance on this?

Reply:
Ok, spent a few hours tonight working on this and I think I have it. Few things to resolve before I can commit though but it's good progress!

Reply:
Got up early today and put in some more work, just have updated to the documentation and then I'll be able to commit these changes

Reply:
Changes are in